### ISSUE TITLE
Assertion 'p = load_fragment_gperf_lookup(i, strlen(i))' failed at ../systemd-stable/src/core/load-fragment.c:4623, function unit_dump_config_items().

### ISSUE THREAD

**Issue by faho** [systemd/systemd#6596]
### Submission type

 - Bug report

### systemd version the issue has been seen with

> systemd 234
+PAM -AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 -SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN default-hierarchy=hybrid

### Used distribution

> Archlinux

Package version is 234.11-6.

### In case of bug report: Expected behaviour you didn't see

> The list of configuration items

### In case of bug report: Unexpected behaviour you saw

> Assertion 'p = load_fragment_gperf_lookup(i, strlen(i))' failed at ../systemd-stable/src/core/load-fragment.c:4623, function unit_dump_config_items().

and a SIGABRT.

### In case of bug report: Steps to reproduce the problem

> `/usr/lib/systemd/systemd --dump-configuration-items`

**Comment by evverx** [comment]
It seems that the bug can be fixed with the following:
```diff
diff --git a/src/core/load-fragment-gperf-nulstr.awk b/src/core/load-fragment-gperf-nulstr.awk
index b52438a..44bc1fb 100644
--- a/src/core/load-fragment-gperf-nulstr.awk
+++ b/src/core/load-fragment-gperf-nulstr.awk
@@ -4,7 +4,7 @@ BEGIN{
         print "const char load_fragment_gperf_nulstr[] ="
 }
 keyword==1 {
-        print "\"" $$1 "\\0\""
+        print "\"" $1 "\\0\""
 }
 /%%/ {
         keyword=1
```
, but I haven't tested it yet.

**Comment by evverx** [comment]
The fix is waiting in #6646.
